src/solvers/dgmulti/dg.jl

```diff
@@ -34,20 +34,63 @@ mul_by_accum!(A, α) = @inline (out, x)->matmul!(out, A, x, α, one(eltype(out))
 @inline each_face_node_global(mesh::AbstractMeshData, dg::DGMulti, cache) = Base.OneTo(dg.basis.Nfq * mesh.md.num_elements)
 
 # interface with semidiscretization_hyperbolic
-wrap_array(u_ode::StructArray, mesh::AbstractMeshData, equations, dg::DGMulti, cache) = u_ode
+wrap_array(u_ode, mesh::AbstractMeshData, equations, dg::DGMulti, cache) = u_ode
 function digest_boundary_conditions(boundary_conditions::NamedTuple{Keys,ValueTypes}, mesh::AbstractMeshData,
                                     dg::DGMulti, cache) where {Keys,ValueTypes<:NTuple{N,Any}} where {N}
   return boundary_conditions
 end
 
-function allocate_coefficients(mesh::AbstractMeshData, equations, dg::DGMulti, cache)
+# Allocate nested array type for DGMulti solution storage.
+function allocate_nested_array(uEltype, nvars, array_dimensions)
+
+  # old approach: store components as separate arrays, combine via StructArrays
+  return StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(uEltype, array_dimensions...), nvars))
+
+  # TODO: any way to use a raw array of size (nvars, array_dimension...) for contiguity in memory?
+  # Can use the following:
+  #   `u_array = zeros(uEltype, nvars, array_dimensions...)`
+  #   `u = StructArray{SVector{nvars, uEltype}}(ntuple(i->view(u_array, i, ntuple(_->:, length(array_dimensions))...), nvars))`
+  # However, then `similar(u)` becomes `StructArray{<:SVector, NTuple{N, <:Matrix}...}`, and
+  # redefining `similar(u)` breaks things.
```

Comment:
If you really want an array using the same memory layout as the other solvers in Trixi, why don't you store it as an `Array{SVector{NVARS, uEltype}}`?

Reply:
This is because I also use the component-wise interpretation - interpolation, projection, and weak form operators require matrix multiplications on individual solution component arrays, which are easy to access using `StructArray`. Moreover, since StructArrays just stores arrays for each solution component and assembles Struct entries on-the-fly, the "old approach" should be fast for `mul!` or `matmul!` from Octavian.

However, I just realized I could specialize the storage for `approximation_type = SBP()` and flux differencing volume integrals, since these don't require the use of matrix multiplication routines. I'll benchmark `Array{SVector{NVARS, uEltype}}` and see if it makes a difference.